# Задача 1.
Контрольная работа состоит из пяти вопросов. На ĸаждый вопрос приведено четыре
варианта ответа, один из ĸоторых правильный. Составьте заĸон распределения числа
правильных ответов при простом угадывании. Найдите M(X) , D(X) .

In [1]:
import numpy as np
from math import factorial

Решение:
n = 5 (пять вопросов), p = 1/4(выбираем 1 ответ из 4)
x = 0 (не угадали ничего),x = 1(угадали 1 ответ) и т.д x = 2, x = 3, x = 4, x = 5.

Найдём распределение Пуансона для каждого события x

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
n = 5
k = 0 #ни один ответ не угадан
p = 1/4
X0 = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X0

0.2373046875

In [4]:
k = 1 # угадан 1 ответ
X1 = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X1

0.3955078125

In [5]:
k = 2 #угадано 2 ответа
X2 = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X2

0.263671875

In [6]:
k = 3 #угадано 3 ответа
X3 = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X3

0.087890625

In [7]:
k = 4 #угадано 4 ответа
X4 = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X4

0.0146484375

In [8]:
k = 5 #все ответы угаданы
X5 = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X5

0.0009765625

Закон распределения правильных ответов выглядит так:

 x     0     1     2     3     4      5
 
 p   0.237 0.396 0.264 0.0879 0.0146 0.000977

Посчитаем математическое ожидание:

M(X)= np = 5⋅0.25 = 1.25

Теперь посчитаем дисперсию, зная, что величина  q (1 - p = 0.75) :

D(X)= npq = 5⋅0.25⋅0.75 = 0.9375

# Задача 2.
Вероятность попадания в цель при одном выстреле равна 0.01. Каĸова вероятность
того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

Решение:
n = 200, p = 0,01, x = 5(5 попаданий), x = 6 (6 попаданий) и т.д. до 10, x = 7, x = 8, x = 9, x = 10
Используем формулу Пуансона

In [9]:
n = 200
m = 5 # 5 попаданий
p = 0.01
lambda_ = n * p

X5 = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)
X5

0.03608940886309672

In [10]:
m = 6 # 6 попаданий
X6 = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)
X6

0.012029802954365574

In [11]:
m = 7 # 7 попаданий
X7 = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)
X7

0.0034370865583901638

In [12]:
m = 8 # 8 попаданий
X8 = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)
X8

0.0008592716395975409

In [13]:
m = 9 # 9 попаданий
X9 = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)
X9

0.000190949253243898

In [14]:
m = 10 # 10 попаданий
X10 = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)
X10

3.8189850648779595e-05

Так как события несовместные, искомая вероятность может быть найдена по формуле сложения вероятностей:
P6(5 ≤ k ≤ 10)= P6(5) + P6(6) + P6(7) + P6(8) + P6(9) + P6(10) =

In [15]:
P6 = X5 + X6 + X7 + X8 + X9 + X10
P6

0.052644709119342686

# Задача3
В условиях предыдущей задачи сĸольĸо выстрелов нужно сделать чтобы быть
уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?
Подсĸазĸа. Каĸ посчитать P(X >= k) для величины X , имеющей распределение
Пуассона? Здесь предстоит немножĸо поĸодить

Решение:
P(X>=k) для велечины X ровна 1 - P(X< k), то количество промахов, при 10 попаданиях будет n - 10, значит наша вероятность равна
0.9 = 1 - q в степени (n-10),q = 1-p, т.е. 1-0.01=0.99. Отсюда получается, что 0.9 = 0.99 в степени (n-10). Найдём n путём подбора.

In [16]:
p = 0.1

for i in range(1, 1600, 1):
    n1 = i
    
    q1 = 0.99 ** n1
    
    if 0 < (p - q1) < 0.001:
        break
print(f" Количество выстрелов равно {n1 + 10}")

 Количество выстрелов равно 240


При ручном подборе получилось 228, значит количество выстрелов 238

В первом случае получили вероятность, числа попаданий от 0 до 10. 

Вариант, как я думаю, верный, представим 0.9 как 1 -(P(10)+P(9)+P(8)+P(7)+P(6)+P(5)+P(4)+P(3)+P(2)+P(1)+P(0)).Путём подбора
получаем количество выстрелов.


In [17]:
p = 0.01
p2 = 0.9
for i in range(100, 10000, 10):
    lambda_ = i * p
    p1 = 1 - ((lambda_**9 / np.math.factorial(9)) * np.exp(-lambda_) +(lambda_**8 / np.math.factorial(8)) * np.exp(-lambda_)+
    (lambda_**7 / np.math.factorial(7)) * np.exp(-lambda_) +(lambda_**6 / np.math.factorial(6)) * np.exp(-lambda_) + 
    (lambda_**5 / np.math.factorial(5)) * np.exp(-lambda_) +(lambda_**4 / np.math.factorial(4)) * np.exp(-lambda_) +
    (lambda_**3 / np.math.factorial(3)) * np.exp(-lambda_) +(lambda_**2 / np.math.factorial(2)) * np.exp(-lambda_) + 
    (lambda_**1 / np.math.factorial(1)) * np.exp(-lambda_)+(lambda_**0 / np.math.factorial(0)) * np.exp(-lambda_))
    if 0 < p2 - p1 < 0.01:
        break
print(i)        

1400
